src/detect.c

```diff
@@ -1790,6 +1893,14 @@ static void DetectRunTx(ThreadVars *tv,
         if (!ires.has_next)
             break;
     }
+
+    /* apply default policy if there were txs to inspect, we have fw rules and non of the rules
+     * applied a policy. */
+    if (tx_inspected && have_fw_rules && !fw_verdict) {
```

Comment:
fw_verdict may be set in tx 0, but then tx 1 might not have eval'd any rules in case of prefiltered rules? Then we should drop for tx 1, but since tx 0 already set this to true we won't? Need a test for this.

Reply:
maybe fw_verdict and tx_inspected both should be counters, and if they aren't equal at the end, we drop